rules/no-reduce.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+const getDocumentationUrl = require('./utils/get-documentation-url');
+
+const message = 'Array.reduce not allowed';
+
+const create = context => {
+	return {
+		'CallExpression > MemberExpression[property.name="reduce"]'(node) {
+			// For arr.reduce()
+			context.report({node: node.property, message});
+		},
+		'CallExpression > MemberExpression[property.name="call"] > MemberExpression[property.name="reduce"]'(node) {
```

Comment:
I think `a(b.reduce)` `a[b.reduce]()` will hit the selector. Can you take a look at `utils/method-selector.js`, should be  better n  that way.